src/routes/login_page.jsx

```diff
@@ -17,54 +18,77 @@ export default class LoginPage extends React.Component {
       isNetworkError: false,
     };
 
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.handleLoginSubmit = this.handleLoginSubmit.bind(this);
+    this.updateField = this.updateField.bind(this);
     this.handleAlertClose = this.handleAlertClose.bind(this);
+    this.handleLogin = this.handleLogin.bind(this);
   }
 
-  handleInputChange(e, field) {
+  updateField(e, field) {
     this.setState({
-      [field]: e.target.value,
+      [field]: e.currentTarget.value,
     });
   }
 
-  handleLoginSubmit(event) {
-    const { username, password } = this.state;
-    event.preventDefault();
-    if (!(username && password)) {
+  async handleLogin(e) {
+    e.preventDefault();
+    const { email, password } = this.state;
+
+    if (!(email || password)) {
       this.setState({
         isAuthenticationError: true,
         isInvalidInput: true,
         isInvalidCredentials: false,
         isNetworkError: false,
       });
-
       return;
     }
 
     this.setState({
-      isAuthenticating: false,
-      isAuthenticated: false,
-      isAuthenticationError: false,
-      isInvalidCredentials: false,
-      isInvalidInput: false,
-      isNetworkError: false,
+      isAuthenticating: true,
     });
 
-    if (username === "ziv" && password === "papa") {
-      this.setState({
-        isAuthenticating: false,
-        isAuthenticated: true,
-        isAuthenticationError: false,
-      });
+    const { data, status } = await authenticateUser({ email, password }).catch(
+      () => {
+        this.setState({
+          isAuthenticating: false,
+          isAuthenticated: false,
+          isInvalidCredentials: false,
+          isInvalidInput: false,
+          isAuthenticationError: true,
+          isNetworkError: true,
+        });
+      }
+    );
+
+    if (status === 200) {
```

Comment:
When you place this in a try/catch expression, the status === 200 check will be redundant since your authenticate.js already has a status === 200 check and otherwise it will throw an error and go into the catch